classes/class-wc-connect-shipping-method.php

```diff
@@ -301,37 +355,19 @@ public function calculate_shipping( $package = array() ) {
 			$predefined_boxes = $this->service_settings_store->get_predefined_packages_for_service( $this->service_schema->id );
 			$predefined_boxes = array_values( array_filter( $predefined_boxes, array( $this, 'filter_preset_boxes' ) ) );
 
-			$response_body = $this->api_client->get_shipping_rates( $services, $package, $custom_boxes, $predefined_boxes );
-
-			if ( is_wp_error( $response_body ) ) {
-				$this->debug(
-					sprintf(
-						'Request failed: %s',
-						$response_body->get_error_message()
-					),
-					'error'
-				);
-				$this->log_error(
-					sprintf(
-						'Error. Unable to get shipping rate(s) for %s instance id %d.',
-						$this->id,
-						$this->instance_id
-					),
-					__FUNCTION__
-				);
-
-				$this->set_last_request_failed();
-
-				$this->log_error( $response_body, __FUNCTION__ );
-				$this->add_fallback_rate( $service_settings );
-				return;
-			}
-
-			if ( ! property_exists( $response_body, 'rates' ) ) {
-				$this->debug( 'Response is missing `rates` property', 'error' );
-				$this->set_last_request_failed();
-				$this->add_fallback_rate( $service_settings );
-				return;
+			$cache_key = sprintf(
+				'wcs_rates_%s',
+				hash( 'md5', serialize( array( $services, $package, $custom_boxes, $predefined_boxes ) ) )
+			);
+			$response_body = wp_cache_get( $cache_key );
+			if ( false !== $response_body ) {
+				$this->debug( 'Rates response retrieved from cache' );
+			} else {
+				$response_body = $this->api_client->get_shipping_rates( $services, $package, $custom_boxes, $predefined_boxes );
+				if ( $this->check_and_handle_response_error( $response_body ) ) {
+					return;
+				}
+				wp_cache_set( $cache_key, $response_body, '', 3600 );
```

Comment:
I'd rather us separate the addition of `/wp-admin` cart detection and shipping rate request caching so they can be tested/vetted in isolation. Can you put these changes in to a new PR?